internal/pkg/filesystem/filesystem.go

```diff
@@ -0,0 +1,330 @@
+// Directories are moved recursively.
+// The destination path must not exist.
+func (f *Fs) Move(src, dst string) error {
+	if f.Exists(dst) {
+		return fmt.Errorf(`cannot move "%s" -> "%s": destination exists`, src, dst)
+	}
+
+	var err error
+	if f.IsFile(src) {
+		if err = f.fs.Rename(src, dst); err != nil {
+			return err
+		}
+	} else {
+		if err = f.Copy(src, dst); err != nil {
+			return err
+		}
+		if err = f.Remove(src); err != nil {
+			return err
+		}
+	}
+
+	f.logger.Debugf(`Moved "%s" -> "%s"`, src, dst)
+	return err
+}
+
+// MoveForce src to dst.
+// Directories are moved recursively.
+// The destination is deleted and replaced if it exists.
+func (f *Fs) MoveForce(src, dst string) error {
+	if f.Exists(dst) {
+		if err := f.Remove(dst); err != nil {
+			return err
+		}
+	}
+	return f.Move(src, dst)
+}
+
+// Remove file or dir.
+// Directories are removed recursively.
+func (f *Fs) Remove(path string) error {
+	err := f.fs.RemoveAll(path)
+	if err == nil {
+		f.logger.Debugf(`Removed "%s"`, path)
+	}
+	return err
+}
+
+// ReadFile content as string.
+func (f *Fs) ReadFile(path, desc string) (*model.File, error) {
+	file := model.CreateFile(path, desc, "")
+
+	// Open
+	fd, err := f.fs.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("missing %s \"%s\": %w", file.Desc, file.Path, err)
+		}
+		return nil, fmt.Errorf("cannot open %s \"%s\": %w", file.Desc, file.Path, err)
+	}
+
+	// Read
+	content, err := ioutil.ReadAll(fd)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read %s \"%s\": %w", file.Desc, file.Path, err)
+	}
+
+	// Close
+	if err := fd.Close(); err != nil {
+		return nil, err
+	}
+
+	f.logger.Debugf(`Loaded "%s"`, path)
+	file.Content = string(content)
+	return file, nil
+}
+
+// WriteFile from string.
+func (f *Fs) WriteFile(file *model.File) error {
+	// Create dir
+	dir := filepath.Dir(string(file.Path))
+	if !f.Exists(dir) {
+		if err := f.Mkdir(dir); err != nil {
+			return err
+		}
+	}
+
+	// Open
+	fd, err := f.fs.OpenFile(string(file.Path), os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("cannot open %s \"%s\": %w", file.Desc, file.Path, err)
+	}
+
+	// Write
+	_, err = fd.WriteString(file.Content)
+	if err != nil {
+		return fmt.Errorf("cannot write %s \"%s\": %w", file.Desc, file.Path, err)
+	}
+
+	// Close
+	if err := fd.Close(); err != nil {
+		return err
+	}
+
+	f.logger.Debugf(`Saved "%s"`, file.Path)
+	return nil
+}
+
+// CreateOrUpdateFile lines.
+func (f *Fs) CreateOrUpdateFile(path, desc string, lines []model.FileLine) (updated bool, err error) {
+	// Read file if exists
+	file, err := f.ReadFile(path, desc)
+	switch {
+	case err != nil && !os.IsNotExist(errors.Unwrap(err)):
+		return false, err
+	case file == nil:
+		updated = false
+		file = model.CreateFile(path, desc, "")
+	default:
+		updated = true
+	}
+
+	// Process expected lines
+	for _, line := range lines {
+		newValue := strings.TrimSuffix(line.Line, "\n") + "\n"
```

Comment:
preco tu odstranujes `\n` a zroven ho aj pridavas konkatenaciou?

Reply:
ahoj, odstrani `\n` AK TAM JE, a VZDY prida `\n` -> tj. subor vzdy bude koncit `\n`, podobne ako ked ulozis subor v IDE.

v Go sa to IMHO neda zapisat kratsie.

inak tento kod sa nemenil iba presunul (v tomto PR je to este duplicitne, na 2 miestach).